lib/supavisor/client_handler.ex

```diff
@@ -173,28 +173,41 @@ defmodule Supavisor.ClientHandler do
 
         Registry.register(Supavisor.Registry.TenantClients, id, [])
 
-        if info.tenant.enforce_ssl and !data.ssl do
-          Logger.error("Tenant is not allowed to connect without SSL, user #{user}")
+        {_type, port} = sock
+        {:ok, addr} = HH.addr_from_port(port)
```

Comment:
the connection will crash if the function returns a non {:ok, _} response

Reply:
This should be okay yeah? 

It will fail closed, users just won't get a nice error message (should see it in the logs still).

I didn't want to rewrite too much of the client handler.

Probably should move to a `with` clause here as we're doing more than 3 checks in nested conditions.